src/DynamoCore/PackageManager/PackageLoader.cs

```diff
@@ -34,13 +39,107 @@ public PackageLoader(string overridePackageDirectory)
                 Directory.CreateDirectory(RootPackagesDirectory);
         }
 
-        private readonly ObservableCollection<Package> localPackages = new ObservableCollection<Package>();
-        public ObservableCollection<Package> LocalPackages { get { return localPackages; } }
+        private void OnPackageAdded(Package pkg)
```

Comment:
Let's remove `ObservableCollection` here and prevent collection mutation from outside the class.